### PR TITLE
Support clash-meta as a systemd service in Fedora 41

### PR DESCRIPTION
**Pull request by hjl-tools**
Add clash-meta.sh, clash-meta.service and config.yaml.sample to support clash-meta as a systemd service in Fedora 41.

### CODE REVIEW

File: clash-meta/clash-meta.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=A rule based proxy in Go.
+After=network.target
+
+[Service]
+Type=exec
+Restart=on-abort
+ExecStart=/usr/bin/clash-meta -d /etc/clash-meta
```
Comment (ImBearChild) [inline clash-meta/clash-meta.service]: `clash-meta` will download file to its configuration directory, such as ip-geolocation database or web UI files.  These files are not strictly "configuration files". Maybe we should put it under `/var/lib/clash-meta`. And clash can use main configuration file from another path. Just like this:

```
ExecStart=/usr/bin/clash-meta -f /etc/clash-meta/config.yaml -d /var/lib/clash-meta
```

File: clash-meta/clash-meta.spec [hunk 2]
```diff
@@ -58,7 +59,11 @@ grep -r '//\s*import\s*"' _build/src/github.com/metacubex/mihomo/vendor/github.c
 %install
 %go_vendor_license_install -c %{S:2}
 install -m 0755 -vd                     %{buildroot}%{_bindir}
+install -m 0755 -vd                     %{buildroot}%{_unitdir}
+install -m 0755 -vd                     %{buildroot}%{_sysconfdir}/clash-meta
 install -m 0755 -vp %{gobuilddir}/bin/* %{buildroot}%{_bindir}/
+install -m 0644 -vp %{S:3}              %{buildroot}%{_unitdir}/
+install -m 0644 -vp docs/config.yaml %{buildroot}%{_sysconfdir}/clash-meta/config.yaml.sample
```
Comment (ImBearChild) [inline clash-meta/clash-meta.spec]: Should we put it under `/usr/share/doc/clash-meta` ?

**Review by ImBearChild — CHANGES_REQUESTED** [review]

File: clash-meta/clash-meta.service (hunk 1 above)
Comment (hjl-tools) [reply clash-meta/clash-meta.service]: Good idea.  I will change.

File: clash-meta/clash-meta.spec (hunk 2 above)
Comment (hjl-tools) [reply clash-meta/clash-meta.spec]: I will fix.

**Pull request merged**